### PR TITLE
Network: Added an executable to host an dedicated room for local wifi

### PR DESCRIPTION
**Pull request by B3n30**
This adds an executable to host a dedicated room for local wifi. Usable as a CLI with this commands:

--room-name             The name of the room
--port                         The port used for the room
--max_members        The maximum number of players for this room
--password                The password for the room
--preferred-game      The preferred game for this room
--preferred-game-id The preferred game-id for this room
--username                The username used for announce
--token                       The token used for announce
--announce-url          The url to the announce server
-h, --help                   Display this help and exit
-v, --version              Output version information and exit

This PR requires #3069 to get merged first

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3071)

### CODE REVIEW

File: src/dedicated_room/citra-room.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            case 'h':
+                PrintHelp(argv[0]);
+                return 0;
+            case 'v':
+                PrintVersion();
+                return 0;
+            }
+        }
+    }
+    std::cout << port << std::endl;
```
Comment (JayFoxRox) [inline src/dedicated_room/citra-room.cpp]: Is this intentional? It's not evident from the output what this number reflects.

File: src/dedicated_room/citra-room.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+              << " Libnetwork: " << Network::network_version << std::endl;
+}
+
+/// Application entry point
+int main(int argc, char** argv) {
+    int option_index = 0;
+    char* endarg;
+#ifdef _WIN32
+    int argc_w;
+    auto argv_w = CommandLineToArgvW(GetCommandLineW(), &argc_w);
```
Comment (Subv) [inline src/dedicated_room/citra-room.cpp]: Do you use this anywhere?

File: src/dedicated_room/citra-room.cpp (hunk 1 above)
Comment (B3n30) [reply src/dedicated_room/citra-room.cpp]: Debug remainder, sorry for that

**Comment by neico** [comment]
Why is `--max_members` different from the other options and uses an underscore instead of a dash?
Also this might be because I'm used to it due to some game engines, but why not name it `--maxplayers` instead? (Seeing as it's description also uses the term player instead of member)

File: src/dedicated_room/citra-room.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        }
+    }
+
+    if (room_name.empty()) {
+        std::cout << "room name is empty!\n\n";
+        PrintHelp(argv[0]);
+        return -1;
+    }
+    if (preferred_game.empty()) {
+        std::cout << "prefered game is empty!\n\n";
```
Comment (acnleditor2) [inline src/dedicated_room/citra-room.cpp]: preferred

File: src/dedicated_room/citra-room.cpp [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        PrintHelp(argv[0]);
+        return -1;
+    }
+    if (preferred_game_id == 0) {
+        std::cout << "prefered-game-id not set!\nThis should get set to allow users to find your "
```
Comment (acnleditor2) [inline src/dedicated_room/citra-room.cpp]: preferred

**Comment by jackmawer** [comment]
> announce-url should default to citra's service,

Excuse me for my ignorance, but what *is* Citra's default announce url? I'm sure I'm being a bit thick, but I can't find it for the life of me!

**Comment by B3n30** [comment]
We can't merge this until #3069 is merged. I have the typos fixed locally, but I'll wait to push it together with changes that might get necessary due to changes in #3069 (there actually are some)

File: src/dedicated_room/citra-room.cpp [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+              << " Libnetwork: " << Network::network_version << std::endl;
+}
+
+/// Application entry point
+int main(int argc, char** argv) {
+    int option_index = 0;
+    char* endarg;
+
+    // This is just to be able to link against core
+    gladLoadGLLoader(static_cast<GLADloadproc>(SDL_GL_GetProcAddress));
```
Comment (wwylele) [inline src/dedicated_room/citra-room.cpp]: Could you elaborate the issue caused by not calling this functions? Can it be resolved if we clean up some dependency mess?

Comment (B3n30) [reply src/dedicated_room/citra-room.cpp]: Will be removed as soon as we separated the announce_web_service stuff from core, so we don't have to link against core.

**Pull request merged**